File: android/src/main/kotlin/com/borda/zebra_rfid_reader_sdk/utils/TagDataResponse.kt
package com.borda.zebra_rfid_reader_sdk.utils

import com.google.gson.Gson

class TagDataResponse {
    private var tagId: String = ""
    private var lastSeenTime: String = ""

    constructor(tagId: String, lastSeenTime: String) {
        this.tagId = tagId
        this.lastSeenTime = lastSeenTime
    }

    fun toJson(): String {
        return Gson().toJson(this)
    }
}

File: ios/Classes/TagDataResponse.swift
import Foundation

class TagDataResponse: Codable {
    var tagId: String
    var lastSeenTime: String

    init(tagId: String, lastSeenTime: String) {
        self.tagId = tagId
        self.lastSeenTime = lastSeenTime
    }

    func toJson() -> String? {
        let encoder = JSONEncoder()
        encoder.outputFormatting = .prettyPrinted
        if let jsonData = try? encoder.encode(self) {
            return String(data: jsonData, encoding: .utf8)
        }
        return nil
    }
}
